backend/src/database/program_classes.go

```diff
@@ -87,6 +87,32 @@ func (db *DB) GetProgramClassDetailsByID(id int, args *models.QueryContext) ([]m
 	return classDetails, nil
 }
 
+type ProgramClassOutcome struct {
+	Month       string `json:"month"`
+	Dropouts    int    `json:"dropouts"`
+	Completions int    `json:"completions"`
+}
+
+func (db *DB) GetProgramClassOutcome(id int, args *models.QueryContext) ([]ProgramClassOutcome, error) {
+	var outcome []ProgramClassOutcome
+
+	query := db.WithContext(args.Ctx).
+		Table("(SELECT TO_CHAR(DATE_TRUNC('month', NOW()) - INTERVAL '1 month' * gs.i, 'YYYY-MM') AS month FROM generate_series(0, 5) AS gs(i)) AS months").
+		Select(`
+        months.month,
+        COALESCE(COUNT(CASE WHEN pce.enrollment_status = 'Completed' THEN 1 END), 0) AS completions,
+        COALESCE(COUNT(CASE WHEN pce.enrollment_status IN ('Incomplete: Dropped', 'Incomplete: Failed to Complete', 'Incomplete: Transferred', 'Incomplete: Withdrawn') THEN 1 END), 0) AS dropouts
+    `).
+		Joins(`LEFT JOIN program_class_enrollments pce ON TO_CHAR(DATE_TRUNC('month', pce.updated_at), 'YYYY-MM') = months.month AND pce.class_id = ?`, id).
```

Comment:
@corypride Did you mention to filter by class_id or program id (`id`)? I'm not getting the right data back because the program id is passed in for the class id filter. 

Additionally, we should take into consideration the user's selected facility so that we show them outcomes depending on the facility-program combination.

Can you break out the `WHERE` statement into it's own clause so that it improves the readability?